src/mpi/spawn/spawn_impl.c

```diff
@@ -122,7 +122,7 @@ int MPIR_Comm_join(int fd, MPI_Comm * intercomm)
     mpi_errno = MPIR_Close_port_impl(local_port);
     MPIR_ERR_CHECK(mpi_errno);
 
-    MPIR_OBJ_PUBLISH_HANDLE(*intercomm, intercomm_ptr->handle);
+    *p_intercomm_ptr = intercomm_ptr;
 
```

Comment:
Why this change? Should we return the handle of `intercomm`?

Reply:
All `MPIR/MPID` functions will use handle pointers (except for a few corner cases that we have to hard code the rules into the script). Remove the corner cases unless we have necessary reasons reduces the complexity that we have to maintain.